### DISCUSSION TITLE
Error: You must authenticate with your public key. Also, make sure withCredentials is enabled for all requests.

### DISCUSSION THREAD

**Discussion by JaberSE09** [swellstores/community#260] [Help] [answered]
When I run the commands for horizon

**Comment by sags95** [comment]
@JaberSE09 Would you be able to detail where/when you're seeing this error?

**Comment by JaberSE09** [comment]
when running npm setup it gives me an error saying [FAILED] Error 1: Cannot query field "content" on type "SwellProduct".
    ✖ GraphQL Doc...

**Reply by sags95** [reply]
This looks like you're referring to Horizon setup. In the case would you be able to pull the latest changes and ensure you have the following env variables.

- `NEXT_PUBLIC_SWELL_STORE_URL`: This endpoint should have the following format: https://{YOUR_STORE_URL}.swell.store
- `NEXT_PUBLIC_SWELL_PUBLIC_KEY`: This value refers to a Store's Public Key that can be found inside the Developer > API Keys route in the Admin Dashboard.
- `SWELL_STORE_ID`: The store's ID. It can be found at the top of the Developer > API Keys route in the Admin Dashboard.
- `SWELL_SECRET_KEY`: A Store's Secret Key. Can be found inside the Developer > API Keys route in the Admin Dashboard, and it's used to sync the store with the local editor configuration.
- `SWELL_STOREFRONT_ID`: The storefront id you are using Horizon with. It can be found at the top of: All Storefronts > your storefront > Developer in the Admin Dashboard.
- `NEXT_PUBLIC_SWELL_EDITOR`: This should be set to true to allow for correct functionality when using the swell editor. In a production environment you can safely set this to false.

 We recently pushed lots of changes the other day which should resolve this error.

**Comment by JaberSE09** [comment]
error ./lib/graphql/client.ts:2:0
Module not found: Can't resolve 'lib/graphql/generated/sdk'
  1 | import { GraphQLClient } from 'graphql-request';
> 2 | import { getSdk } from 'lib/graphql/generated/sdk'; // THIS FILE IS THE GENERATED FILE

**Comment by JaberSE09** [comment]
when run npm run dev

**Reply by sags95** [reply]
In `package.json` can you let me know what version of `@swell/cli` is being installed?

**Comment by JaberSE09** [comment]
"@swell/cli": "^1.0.2",

**Comment by JaberSE09** [comment]
The full error is 
exec error: Error: Command failed: swell model push -c -f ./config/content/category.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID
node:fs:1026
  handleErrorFromBinding(ctx);
  ^

Error: EPERM: operation not permitted, rename '.swell\config.json.1966938972' -> '.swell\config.json'
    at Object.renameSync (node:fs:1026:3)
    at Function.writeFileSync [as sync] (/node_modules/@swell/cli/dist/bin/swell-0f71f07d.js:23785:8)
    at Configstore.set all [as all] (node_modules/@swell/cli/dist/bin/swell-0f71f07d.js:24148:20)
    at new Configstore (node_modules/@swell/cli/dist/bin/swell-0f71f07d.js:24112:13)
    at node_modules/@swell/cli/dist/bin/swell-0f71f07d.js:24205:21
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:526:24)
    at async loadESM (node:internal/process/esm_loader:91:5)
    at async handleMainPromise (node:internal/modules/run_main:65:12) {
  errno: -4048,
  syscall: 'rename',
  code: 'EPERM',
  path: .swell\\config.json.1966938972',
  dest: .swell\\config.json'
}

exec error: Error: Command failed: swell model push -c -f ./config/content/divider.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell storefronts push --id $SWELL_STOREFRONT_ID -t settings -f ./config/defaults.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell storefronts toggle-editor --id $SWELL_STOREFRONT_ID --theme-id horizon --enable true --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell model push -c -f ./config/content/heading_with_text.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell model push -c -f ./config/content/multiple_panels.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell model push -c -f ./config/content/products_preview.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell model push -c -f ./config/content/quiz.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell model push -c -f ./config/content/press_mentions_carousel.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell model push -c -f ./config/content/multiple_features.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell model push -c -f ./config/content/memberships.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell model push -c -f ./config/content/categories_preview.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell model push -c -f ./config/content/full_width_media.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell model push -c -f ./config/content/quiz_question.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell model push -c -f ./config/content/quiz_transition.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell model push -c -f ./config/content/reviews.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell model push -c -f ./config/content/page.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell model push -c -f ./config/content/product.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID

exec error: Error: Command failed: swell storefronts push --id $SWELL_STOREFRONT_ID -t editor -f ./config/editor.json --secret-key $SWELL_SECRET_KEY --store $SWELL_STORE_ID


> @swell/horizon@0.4.0 graphql:generate
> graphql-codegen --config codegen.yml -r dotenv/config

✔ Parse Configuration
⚠ Generate outputs
  ❯ Generate to ./graphql.schema.json
    ✔ Load GraphQL schemas
    ✔ Load GraphQL documents
    ✖ GraphQL Document Validation failed with 2 errors;
      Error 0: Cannot query field "content"…
  ❯ Generate to lib/graphql/generated/sdk.ts
    ✔ Load GraphQL schemas
    ✔ Load GraphQL documents
    ✖ GraphQL Document Validation failed with 2 errors;
      Error 0: Cannot query field "content

**Reply by sags95** [reply]
In this case it looks like a Windows permission issue. Can you try running `npm cache clean`, if this doesn't work there might be some other solutions.

**Reply by sags95** [reply]
Can you ensure that `~/.swell` has both read and write permissions? It seems that in this case it's lacking write permissions.

**Comment by JaberSE09** [comment]
I ran that and it did not work is it some thing to do with swell cli because it will not let me login with swell login

**Reply by sags95** [reply]
Outside of horizon when you run `swell login` you get the same error? If you can also keep your responses in the same thread, this will help other members of the team assist you.

**Reply by JaberSE09** [reply]
I get authorization failed

**Reply by logeshswell** [reply, marked as answer]
@JaberSE09 It looks like a permission issue in your Windows machine. Can you try one of the solutions mentioned below:
https://stackoverflow.com/a/45840283

If none of the solutions helps, please share the loom recording of the sequence of actions you perform. I will try to reproduce the same and propose you an answer.
